pkg/workflows/workflow.go

```diff
@@ -301,149 +301,142 @@ func (w *workflowLifecycle) configureWorkflowArtifacts(wf *unstructured.Unstruct
 		return err
 	}
 	for _, template := range templates.([]interface{}) {
+		// Process templates with resource
+		err := w.processWorkflowResources(template, wt)
+		if err != nil {
+			return err
+		}
+
 		if allSteps, found, err := unstructured.NestedFieldNoCopy(template.(map[string]interface{}), "steps"); found {
 			for _, steps := range allSteps.([]interface{}) {
 				steps := steps.([]interface{})
 				for _, step := range steps {
-					err := w.processArtifacts(step, wt)
+					err := w.processWorkflowResources(step, wt)
 					if err != nil {
 						return err
 					}
 				}
 			}
-		} else {
+		} else if err != nil {
 			return err
 		}
 	}
 
 	return nil
 }
 
-func (w *workflowLifecycle) processArtifacts(workflowStepObject interface{}, wt *addonmgrv1alpha1.WorkflowType) error {
-	artifacts, _, err := unstructured.NestedFieldNoCopy(workflowStepObject.(map[string]interface{}), "arguments", "artifacts")
+func (w *workflowLifecycle) processWorkflowResources(workflowStepObject interface{}, wt *addonmgrv1alpha1.WorkflowType) error {
+	artifacts, foundArtifacts, err := unstructured.NestedFieldNoCopy(workflowStepObject.(map[string]interface{}), "arguments", "artifacts")
 	if err != nil {
 		return err
 	}
 
-	if artifacts == nil {
-		return nil
-	}
-
-	for _, artifact := range artifacts.([]interface{}) {
-		artifact := artifact.(map[string]interface{})
-		data, _, _ := unstructured.NestedString(artifact, "raw", "data")
-		resources := strings.Split(data, "---\n")
-
-		var newData []string
-		for _, obj := range resources {
-			resource := make(map[string]interface{})
-			err := yaml.Unmarshal([]byte(obj), &resource)
+	if foundArtifacts {
+		for _, artifact := range artifacts.([]interface{}) {
+			artifact := artifact.(map[string]interface{})
+			data, _, err := unstructured.NestedString(artifact, "raw", "data")
 			if err != nil {
-				return fmt.Errorf("unable to unmarshall artifact: %s", obj)
```

Comment:
you removed the logging.  make sure you can catch the exact error at which step laster.

Reply:
This is not log, just error but now will bubble up the underlying error as it was obscured before.